keras/ops/core_test.py

```diff
@@ -215,6 +215,17 @@ def test_slice_update(self):
         outputs = core.slice_update(inputs, start_indices, updates)
         self.assertAllClose(outputs[1:3, 1:3, 2:4, 2:4], np.zeros([2, 2, 2, 2]))
 
+
+    def test_scan(self):
```

Comment:
Let's use `jax.scan` as the reference and let's test consistency with the reference across multiple inputs and dtypes. Let's also test static shape inference.

Reply:
I have added two tests one with int and one with float. The float is not passing due to rounding errors. Should I round it to remove such errors . Also I am not sure about what do you mean by static shape inference sorry.

Reply:
> Also I am not sure about what do you mean by static shape inference sorry.

This means testing the output shape and dtypes obtained when calling the op on a KerasTensor. There is a test case for this in this file.

Reply:
The test class is called `CoreOpsStaticShapeTest` and is right above this one. You can take a look at some of the test cases for the class to get an idea of how static shape inference testing works.